File: backend/src/main/scala/microservices/flight/objects/FlightDailyLowestPricesPlannerResponse.scala
// FlightDailyLowestPricesPlannerResponse defines the flight daily-lowest-prices planner response model.
package com.typesafe.travel.flight.objects

import io.circe.{Decoder, Encoder}
import io.circe.generic.semiauto.{deriveDecoder, deriveEncoder}

final case class FlightDailyLowestPricesPlannerResponse(
    prices: List[FlightDailyLowestPricePlannerResponse]
)

object FlightDailyLowestPricesPlannerResponse:
  given sourceEncoder: Encoder[FlightDailyLowestPricesPlannerResponse] = deriveEncoder
  given sourceDecoder: Decoder[FlightDailyLowestPricesPlannerResponse] = deriveDecoder

File: frontend/src/microservices/flight/objects/FlightDailyLowestPricesPlannerResponse.ts
// 本文件定义 flight 模块的 `FlightDailyLowestPricesPlannerResponse`，作为价格列表响应数据并提供 JSON 编解码。

import type { FlightDailyLowestPricePlannerResponse } from './FlightDailyLowestPricePlannerResponse'

export type FlightDailyLowestPricesPlannerResponse = {
  prices: FlightDailyLowestPricePlannerResponse[]
}

export const flightDailyLowestPricesPlannerResponseFromJson = (json: string): FlightDailyLowestPricesPlannerResponse =>
  JSON.parse(json) as FlightDailyLowestPricesPlannerResponse

export const flightDailyLowestPricesPlannerResponseToJson = (value: FlightDailyLowestPricesPlannerResponse): string =>
  JSON.stringify(value)

